### ISSUE TITLE
Condition causing AutoMapper to miss string properties

### ISSUE THREAD

**Issue by YumsTheGreat** [AutoMapper/AutoMapper#1702]
I'm experiencing an issue using the following condition when mapping to an existing destination object

```csharp
cfg.CreateMap<ScriptDto, Script>()
   .ForMember(y => y.Id, opt => opt.Ignore())
   .ForAllOtherMembers(opt =>
   {
      opt.Condition((src, dest, srcValue) =>
      {
         if (srcValue != null && srcValue is DateTime)
         {
            if ((DateTime)srcValue <= new DateTime())
            {
               return false;
             }
             return true;
          }
          return srcValue != null;
       });
   });

```

Basically if I use a condition like this my string properties are no longer mapped to one another even though they are named the same. With the condition commented out my data mapped the way I would expect. I would assume the same issue would also happen with other simple datatypes but I haven't had a chance to test it.

I'm using version 5.1.1 in net core.

**Comment by YumsTheGreat** [comment]
I quick update about this. It also appears that all my nullable properties get defaulted when the condition is not met. I was only able to get around the date time being nulled by using the above logic.

**Comment by lbargaoanu** [comment]
Try the MyGet build.

**Comment by lbargaoanu** [comment]
This can be closed.

**Comment by YumsTheGreat** [comment]
So I tested the latest build on myget and it resolves the skipping properties issue so that's good. 

That said, the complex properties in my destination object are still being set to default values after my condition ins't met. For example, when mapping my dto to my existing entity, if I have a DateTime? that is null, when it fails the condition, my destination DateTime property is still set to a default value instead of being skipped. Is mapping a nullable type to a standard type not possible in automapper? 

Ideally what I would like to do is have my json objects (from my REST API) serialized into my dto then update an existing object where the value isn't set to null. In order to do that I need to be able to map nullable types to regular types (for example DateTime? -> DateTime).

Should I open a new issue for this?

**Comment by lbargaoanu** [comment]
If the mapping condition is met, the destination member is assigned, otherwise no. If you see smth else, open a new issue with a failing test, a complete example, not like the one above.

**Comment by YumsTheGreat** [comment]
Okay I will do that when I get a chance.

**Comment by lbargaoanu** [comment]
Maybe you want smth different than conditions, but I cannot tell. A value type always has a value, even if it's not explicitly assigned.

**Comment by YumsTheGreat** [comment]
I think we are misunderstanding each other. In any case I figured out the root of the issue. When I have a nullable property on my source object (like DateTime?), when that property value is being passed to the conditional function it is cast as its non-nullable and thus the condition comparitor is evaluated to true and my source property is overwritten. 

So for example if I have a property

`DateTime? someProperty = null`

on my source object the source value of that property, which should be null, is passed to the conditional function as its non-nullable DateTime default and this my destination property is overwritten because the condition evaluates to true (srcValue != null). That is not the functionality I would expect. I would expect my source property to maintain its "null-ness" when passed to the conditional function, which it is not. Since I'm trying to pass a nullable property on my DTO to a non-nullable on my destination my best guess as to what is happening is the source value is being cast before the conditional function is called. I don't know if this is what you are intending or not.

If you would like I can put my full example in a new open issue since this is kind of diverting from the original issue which has been resolved.
